### ISSUE TITLE
Internal error in Agda.TypeChecking.Rules.Term

### ISSUE THREAD

**Issue by nad** [agda/agda#3403]
The following code triggers internal errors for Agda 2.5.4, 2.5.4.1, 2.5.4.2, and 2.6.0-3161248:
```agda
open import Agda.Builtin.Equality

record Σ (A : Set) (B : A → Set) : Set where
  constructor _,_
  field
    fst : A
    snd : B fst

map : {A B : Set} {P : A → Set} {Q : B → Set} →
      (f : A → B) → (∀ {x} → P x → Q (f x)) →
      Σ A P → Σ B Q
map f g (x , y) = (f x , g y)

postulate
  F         : Set → Set → Set
  apply     : {A B : Set} → F A B → A → B
  construct : {A B : Set} → (A → B) → F A B
  A         : Set
  B         : A → Set
  f         : A → A
  g         : ∀ {x} → B x → B (f x)

k : Σ A B → Σ A _

h : F (Σ A B) (Σ A B)

k = map f g

P : ∀ {x y} → k x ≡ k y → Set₁
P refl = Set

h = construct k
```
For Agda 2.6.0-3161248 I get the following error:
```
An internal error has occurred. Please report this as a bug.
Location of the error: src/full/Agda/TypeChecking/Rules/Term.hs:679
```

**Comment by nad** [comment]
Bisection points to 70e0ff497b732677d2e14aecc94d26e812327b51: "[ fixed #2056 ] Implement postponing of checking a definition by pattern matching".

**Comment by andreasabel** [comment]
Fixed by removing the `__IMPOSSIBLE__` and replacing it by `reraise` (of the caught exception).

**Comment by asr** [comment]
The fix hasn't been back-ported to 2.5.4.3.

**Comment by asr** [comment]
Back-ported into 2.5.4.3.
